components/timezone/src/calculator.rs

```diff
@@ -2,42 +2,47 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use crate::provider::{MetazoneId, TimeZoneBcp47Id};
+use crate::provider::{MetazoneId, TimeZoneBcp47Id, ZoneOffsetPeriodV1Marker};
 
 use crate::provider::MetazonePeriodV1Marker;
+use crate::UtcOffset;
 use icu_calendar::DateTime;
 use icu_calendar::Iso;
 use icu_provider::prelude::*;
 use zerovec::ule::AsULE;
 
-/// [`MetazoneCalculator`] uses data from the [data provider] to calculate metazone id.
+/// [`TimeZoneCalculator`] uses data from the [data provider] to compute [`ResolvedTimeZone`](crate::ResolvedTimeZone)s.
 ///
 /// [data provider]: icu_provider
 #[derive(Debug)]
-pub struct MetazoneCalculator {
+pub struct TimeZoneCalculator {
```

Comment:
Discussion: Not sure about the name `TimeZoneCalculator`. I would expect such a name to be reserved for a TZDB calculator.